### PR TITLE
Me0007ys ultrasonic distance sensor driver

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using Meadow.Devices;
+using Meadow.Foundation.Sensors.Distance;
+using Meadow.Units;
+using System;
+using System.Threading.Tasks;
+
+namespace Me007ys_Sample
+{
+    // Change F7FeatherV2 to F7FeatherV1 for V1.x boards
+    public class MeadowApp : App<F7CoreComputeV2>
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:11]: You want to leave it using a CCM?

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        Me007ys me007ys;
+
+        public override Task Initialize()
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:15]: This sample needs SNIP/SNOP

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+﻿MonoControl:
+    Options: --jit
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml:2]: You can remove this file if you want, JIT is enable by default now.

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            me007ys.Subscribe(consumer);
+
+            me007ys.DistanceUpdated += Me007y_DistanceUpdated;
+
+            return Task.CompletedTask;
+        }
+
+        public override async Task Run()
+        {
+            var distance = await me007ys.Read();
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:44]: Shouldnt this be calling `MeasureDistance`? I dont see a Read method, must be on ByteCommSensorBase

**Review by jorgedevs — CHANGES_REQUESTED** [review]

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs (hunk 1 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:11]: oh right - I'll change :)

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:15]: great catch

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml (hunk 3 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml:2]: will do

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs (hunk 4 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:44]: yeah its on the base class

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml (hunk 3 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/meadow.config.yaml:2]: actually ... I just leave it and we can do a bulk remove ... all of the samples have it currently

File: Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs (hunk 4 above)
Comment (jorgedevs) [reply Source/Meadow.Foundation.Peripherals/Sensors.Distance.Me007ys/Samples/Me007ys_Sample/MeadowApp.cs:46]: Then why do we need a `MeasureDistance` method? Seems redundant.

**Review by jorgedevs — APPROVED** [review]

**Pull request merged**